backend/src/stwo/circuit_builder.rs

```diff
@@ -80,14 +92,29 @@ impl<T: FieldElement> FrameworkEval for PowdrEval<T> {
     }
     fn evaluate<E: EvalAtRow>(&self, mut eval: E) -> E {
         assert!(
-            self.analyzed.constant_count() == 0 && self.analyzed.publics_count() == 0,
-            "Error: Expected no fixed columns nor public inputs, as they are not supported yet.",
+            self.analyzed.publics_count() == 0,
+            "Error: Expected no public inputs, as they are not supported yet.",
         );
 
         let witness_eval: BTreeMap<PolyID, [<E as EvalAtRow>::F; 2]> = self
             .witness_columns
             .keys()
-            .map(|poly_id| (*poly_id, eval.next_interaction_mask(0, [0, 1])))
+            .map(|poly_id| {
+                (
+                    *poly_id,
+                    eval.next_interaction_mask(ORIGINAL_TRACE_IDX, [0, 1]),
+                )
+            })
+            .collect();
+        let constant_eval: BTreeMap<PolyID, <E as EvalAtRow>::F> = self
+            .constant_columns
+            .keys()
+            .map(|poly_id| {
+                (
+                    *poly_id,
+                    eval.get_preprocessed_column(PreprocessedColumn::Plonk(3)),
```

Comment:
Why 3?

Reply:
It can be random number, stwo uses this number to mark how many preprocessed columns are added in [plonk](https://github.com/starkware-libs/stwo/blob/77de4d79b08dd89600994e4dc3dc06f77ddce25b/crates/prover/src/examples/plonk/mod.rs#L52) example. 
[here](https://github.com/starkware-libs/stwo/blob/77de4d79b08dd89600994e4dc3dc06f77ddce25b/crates/prover/src/constraint_framework/mod.rs#L80) shows the parameter PreprocessedColumn is not used actually 
and [here](https://github.com/starkware-libs/stwo/blob/77de4d79b08dd89600994e4dc3dc06f77ddce25b/crates/prover/src/constraint_framework/preprocessed_columns.rs#L11) is the definition of PreprocessedColumn, no declaration of what the number means.